### PR TITLE
Signup

### PR DESCRIPTION
**Pull request by RobertoCarrilloAvila**
# Description

Add SignUp component
[Figma Design](https://www.figma.com/file/Hlq710GJ4jCPU9EDgQS1Dx/Target-Web?node-id=0%3A700&t=fFAxxJ5y7zrNEXt0-0)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots:
IMG_61CC60012806-1
IMG_5149177CB252-1

### CODE REVIEW

File: src/components/SignUp/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+const SignUp = () => {
+  const GENDER_OPTIONS = ["male", "female"];
+
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordConfirmation, setPasswordConfirmation] = useState("");
+  const [gender, setGender] = useState("");
+  const [redirect, setRedirect] = useState(false);
```
Comment (marcos09) [inline src/components/SignUp/SignUp.js:20]: I recommend you use a form validation library like Zod Validations or React hook form instead of these states.

**Review by marcos09 — CHANGES_REQUESTED** [review]
Great job, in general it looks good to me but I leave a couple of comments. 

Another thing you could try to add is internationalization (Maybe in other PR later) It's not mandatory for this project but always it's useful to have it because you don't know when your project could be needed in other languages

File: src/components/SignUp/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    setPassword(value);
+  };
+
+  const handlePasswordConfirmationChange = (value) => {
+    setPasswordConfirmation(value);
+  };
+
+  const handleGenderChange = (value) => {
+    setGender(value);
+  };
```
[the remark this answers is not in the document]
Comment (RobertoCarrilloAvila) [reply src/components/SignUp/SignUp.js]: Done!

File: src/components/SignUp/SignUp.js (hunk 1 above)
Comment (RobertoCarrilloAvila) [reply src/components/SignUp/SignUp.js:20]: I'm going to implement the library in the next PR

File: src/components/SignUp/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+          <h1 id="sign-up-title">Sign Up</h1>
+  
+          <form className="sign-up-form" onSubmit={handleSubmit}>
+            <FormInput
+              type="text"
+              id="name"
+              name="name"
+              label="name"
+              value={name}
+              onChange={(value) => setName(value)}
```
Comment (marcos09) [inline src/components/SignUp/SignUp.js:65]: You could make it easy only sending setName in onChange prop
```suggestion
              onChange={setName}
```

File: src/components/SignUp/SignUp.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+              required={true}
+            />
+  
+            <FormInput
+              type="email"
+              id="email"
+              name="email"
+              label="email"
+              value={email}
+              onChange={(value) => setEmail(value)}
```
Comment (marcos09) [inline src/components/SignUp/SignUp.js:75]: You could make it easy only sending setName in onChange prop

```suggestion
              onChange={setEmail}
```

File: src/components/SignUp/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            />
+  
+            <FormInput
+              type="password"
+              id="password"
+              name="password"
+              label="password"
+              placeholder="min. 6 characters long"
+              value={password}
+              onChange={(value) => setPassword(value)}
```
Comment (marcos09) [inline src/components/SignUp/SignUp.js:86]:
```suggestion
              onChange={setPassword}
```

File: src/components/SignUp/SignUp.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+              required={true}
+            />
+  
+            <FormInput
+              type="password"
+              id="password_confirmation"
+              name="password_confirmation"
+              label="confirm password"
+              value={passwordConfirmation}
+              onChange={(value) => setPasswordConfirmation(value)}
```
Comment (marcos09) [inline src/components/SignUp/SignUp.js:97]:
```suggestion
              onChange={setPasswordConfirmation}
```

File: src/components/SignUp/SignUp.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            />
+  
+            <FormSelect
+              id="gender"
+              name="gender"
+              label="gender"
+              options={GENDER_OPTIONS}
+              placeholder="select your gender"
+              value={gender}
+              onChange={(value) => setGender(value)}
```
Comment (marcos09) [inline src/components/SignUp/SignUp.js:109]:
```suggestion
              onChange={setGender}
```

**Review by marcos09 — CHANGES_REQUESTED** [review]
You could make it easy only sending setState in onChange prop

File: src/components/SignUp/SignUp.js (hunk 7 above)
Comment (RobertoCarrilloAvila) [reply src/components/SignUp/SignUp.js]: Done!

**Review by marcos09 — APPROVED** [review]

**Pull request merged**